hs-abci-sdk/src/Tendermint/SDK/Modules/Validators/Types.hs

```diff
@@ -0,0 +1,59 @@
+module Tendermint.SDK.Modules.Validators.Types where
+
+import           Control.Lens                           (Wrapped (_Wrapped'),
+                                                         iso, (^.), _Unwrapped')
+import qualified Data.Aeson                             as A
+import           Data.Bifunctor                         (Bifunctor (bimap, second))
+import           Data.ByteString                        (ByteString)
+import           Data.ByteString.Lazy                   (toStrict)
+import           Data.Either                            (fromRight)
+import           Data.ProtoLens                         (decodeMessage,
+                                                         encodeMessage)
+import           Data.Set                               (Set)
+import           Data.Text                              (pack)
+import           GHC.Generics                           (Generic)
+import           Network.ABCI.Types.Messages.FieldTypes (PubKey (PubKey),
+                                                         ValidatorUpdate)
+import           Tendermint.SDK.BaseApp                 (RawKey (..))
+import           Tendermint.SDK.Codec                   (HasCodec (..))
+
+
+data ValidatorsNameSpace
+
+type ValidatorsName = "validators"
+
+
+updatesListKey :: ByteString
+updatesListKey = "updatesList"
+
+validatorsMapKey :: ByteString
+validatorsMapKey = "validatorsMap"
+
+validatorsKeySetKey :: ByteString
+validatorsKeySetKey = "validatorsKeySet"
+
+
+newtype ValidatorUpdate_ = ValidatorUpdate_ ValidatorUpdate deriving (Eq, Generic)
+
+instance HasCodec ValidatorUpdate_ where
+  encode (ValidatorUpdate_ vu) = encodeMessage $ (vu ^. _Wrapped')
+  decode bs = bimap pack (ValidatorUpdate_ . (^. _Unwrapped')) $ decodeMessage bs
```

Comment:
We don't really use things like `pack` because it leads to having to import like 10 different `String`-like modules throughout the project. We use the package [string-conversions](https://hackage.haskell.org/package/string-conversions) to abstract it away via the [cs method](https://hackage.haskell.org/package/string-conversions-0.4.0.1/docs/Data-String-Conversions.html#v:cs).

Reply:
:+1: good to know, I too have been annoyed by keeping track of various string types, so having a way to handle that is nice.